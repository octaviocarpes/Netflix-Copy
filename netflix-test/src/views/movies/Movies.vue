<template>
  <div class="movies-container">
    <Navbar />
    <div v-if="!!movies" class="movie-suggestion">
      <MovieSuggestion :movie="trailers ? trailers[0] : {}" />
    </div>
    <div v-if="!!movies" class="movies-collection">
      <h1>Trailers</h1>
      <ul>
        <li v-for="movie in trailers" :key="movie.id">
          <MovieThumbnail :movie="movie" />
        </li>
      </ul>

      <h1>Movies</h1>
      <ul>
        <li v-for="movie in movies" :key="movie.id">
          <MovieThumbnail :movie="movie" />
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
import Navbar from "@/components/navbar/Navbar";
import MovieSuggestion from "@/components/movie-suggestion/MovieSuggestion";
import MovieThumbnail from "@/components/movie-thumbnail/MovieThumbnail";
import store from "@/store";

export default {
  name: "Movies",

  components: {
    Navbar,
    MovieSuggestion,
    MovieThumbnail
  },

  mounted() {
    store.dispatch("GET_MOVIES");
    store.dispatch("GET_TRAILERS");
  },

  computed: {
    movies() {
      return store.state.movies.movies;
    },

    trailers() {
      return store.state.movies.trailers;
    }
  }
};
</script>

<style lang="scss" scoped>
@import "./style.scss";
</style>
