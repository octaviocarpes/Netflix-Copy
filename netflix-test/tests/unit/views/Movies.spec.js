import Vue from "vue";
import Movies from "@/views/movies/Movies.vue";

describe("Movies.spec.js", () => {
  let cmp, vm;

  beforeEach(() => {
    cmp = Vue.extend(Movies);
    vm = new cmp({}).$mount();
  });

  it("has the expected html structure", () => {
    expect(vm.$el).toMatchSnapshot();
  });
});