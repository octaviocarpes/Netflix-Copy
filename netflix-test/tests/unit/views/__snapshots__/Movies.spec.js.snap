// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies.spec.js has the expected html structure 1`] = `
<div
  class="movies-container"
>
  <nav>
    <div
      class="nav-bar-container"
    >
      <div
        class="nav-bar-content-left"
      >
        <div
          class="header-logo"
        >
          <h1>
            NIETFLIX
          </h1>
        </div>
         
        <div
          class="links"
        >
          <ul>
            <li>
              <router-link
                to="movies"
              >
                Movies
              </router-link>
            </li>
             
            <li>
              <router-link
                to="profile"
              >
                Profile
              </router-link>
            </li>
             
            <li>
              <router-link
                to="metrics"
              >
                Metrics
              </router-link>
            </li>
          </ul>
        </div>
      </div>
       
      <div
        class="nav-bar-content-right"
      >
        <div
          class="user"
        >
          <img
            alt="profile icon"
            src=""
          />
           
          <p>
            
          </p>
        </div>
         
        <p>
          Logout
        </p>
      </div>
    </div>
  </nav>
   
  <div
    class="movie-suggestion"
  >
    <!---->
  </div>
   
  <div
    class="movies-collection"
  >
    <h1>
      Trailers
    </h1>
     
    <ul />
     
    <h1>
      Movies
    </h1>
     
    <ul />
  </div>
</div>
`;
